olive/passes/olive_pass.py

```diff
@@ -119,38 +126,50 @@ def _init_fixed_and_search_params(
         Get the fixed and search parameters from the config.
         """
         default_config = self.default_config()
+        param_order = order_search_parameters(config)
 
         # fixed parameters
         fixed_params = {}
-        for key, value in config.items():
-            if isinstance(value, SearchParameter):
-                continue
-            if default_config[key].is_path and value is not None:
-                value = str(Path(value).resolve())
-            fixed_params[key] = value
-
-        # search parameters
         search_space = {}
-        for key, value in config.items():
+        for key in param_order:
+            value = config[key]
+            if isinstance(value, SearchParameter):
+                # resolve conditional parameters
+                # if categorical with single choice, use that choice directly
+                value = self._resolve_search_parameters(value, fixed_params)
+            if value == SpecialParamValue.INVALID:
+                # TODO: better error message, e.g. what the parent values were, how it was invalid
+                raise ValueError(
+                    f"Invalid value for parameter '{key}'. Either the parameter or its parents are not fixed."
+                )
             if isinstance(value, SearchParameter):
-                search_space[key] = self._resolve_search_parameter(value, fixed_params)
+                search_space[key] = value
```

Comment:
> search_space[key] = value [](http://example.com/codeflow?start=15&length=26)

should it be put in above if clause?

Reply:
The first one does the simplification of `Categorical`, `Conditional` and `ConditionalDefault`. All of these can became fixed parameters if 1) Categorical has only one option, 2) Conditional has all parents fixed and value is categorical with one option, 3) ConditionalDefault has all parents fixed so its value is also fixed. 

We only add to `search_space` if the above simplified value is still a search parameter. Otherwise, it is a fixed parameter.